src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -67,8 +67,13 @@ export const RoomEditModal: React.FC<RoomEditModalProps> = ({
   );
 
   const onSubmit = useCallback(() => {
-    return submitHandler(values, room.roomIndex!);
-  }, [room.roomIndex, submitHandler, values]);
+    if (typeof room.roomIndex !== "number") return;
```

Comment:
In what cases will this not be a `number`? Is it only when it's `undefined`? Can it ever be the case? What happens when we 'silently return' here?

I feel like we should be signalling an error in some way or another if this isn't handled some other way, either directly to the client UI, through `Bugsnag.notify`, or potentially both (depending on what the 'solution' to the error is)

Reply:
Well, (property) RoomData_v2.roomIndex?: number | undefined and we should check if it isn't a number.
 Actually, it's just a check for TS cause this case (`roomIndex` is undefined) can't be alive (when you edit room - you will definitely have roomIndex)

Reply:
In that case, we could probably just do `if (room.roomIndex === undefined) return;` to make it more explicit.

Reply:
fixed